app/vmagent/remotewrite/remotewrite.go

```diff
@@ -386,8 +385,8 @@ func Stop() {
 // PushDropSamplesOnFailure drops wr samples if they cannot be sent to -remoteWrite.url by any reason.
 //
 // PushDropSamplesOnFailure can modify wr contents.
-func PushDropSamplesOnFailure(at *auth.Token, wr *prompbmarshal.WriteRequest) {
-	_ = tryPush(at, wr, true)
+func PushDropSamplesOnFailure(at *auth.Token, wr *prompbmarshal.WriteRequest, labelsLimits func([]prompbmarshal.Label) bool) {
```

Comment:
This API doesn't look good to me. It's better to keep it as is and use global check ` func tryPush`.

Reply:
don't know how to override maxLabelsPerTimeseries with a scrape config's `labels_limit` value

Reply:
WDYT? It should be applied at the same place as other scraping limits at `scrapeInternal` function. 

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/lib/promscrape/scrapework.go#L418

Reply:
Also, it should NOT be the part of this pull request.

Reply:
> It should be applied at the same place as other scraping limits at scrapeInternal function.

labels_limit should override maxLabelsPerTimeseries flag value, so at least tryPush should accept labelsLimit function, where PushDropSamplesOnFailure should pass nil